tests/credential_plugins_test.py

```diff
@@ -134,6 +135,75 @@ def test_hashivault_handle_auth_not_enough_args() -> None:
         hashivault.handle_auth()
 
 
+@pytest.mark.parametrize(
+    'kwargs', (
+        (
+            {
+                'access_key': 'my_access_key',
+                'secret_key': 'my_secret_key',
+                'role_arn': 'the_arn',
+                'identifier': 'access_token',
+            },
+        ),
+        (
+            {
+                'role_arn': 'the_arn',
+                'identifier': 'access_token',
+            },
+        ),
+    ),
+)
+@pytest.mark.parametrize(
+    (
+        'identifier_key',
+        'expected',
+    ),
+    (
+        None,
+        'the_access_token',
+    ),
+    (
+        'access_key',
+        'the_access_key',
+    ),
+    (
+        'secret_key',
+        'the_secret_key',
+    ),
```

Comment:
And this one is not nested enough (causing that cryptic exception in pytest and pylint invocations). Let's see if this helps:
```suggestion
    (
	    (None, 'the_access_token'),
	    ('access_key', 'the_access_key'),
	    ('secret_key', 'the_secret_key'),
    ),
```